tests/FSharp.Compiler.ComponentTests/CompilerService/AsyncMemoize.fs

```diff
@@ -18,6 +18,12 @@ let waitFor (mre: ManualResetEvent) =
     if not <| mre.WaitOne timeout then 
         failwith "waitFor timed out"
 
+let waitUntil condition value =
+    let sw = Stopwatch.StartNew()
+    while not <| condition value do
+        if sw.Elapsed > timeout then
+            failwith "waitUntil timed out"
+        Thread.Sleep 10
```

Comment:
This will lock the thread, do we care here?

Reply:
Probably not, but since we're calling this from `task` expression anyway might as well make it async.